test/createvolume_tester.py

```diff
@@ -1,122 +1,146 @@
+
+import hpe_docker_unit_test as hpedockerunittest
+
+
+class CreateVolumeUnitTest(hpedockerunittest.HpeDockerUnitTestExecutor):
+
+    # This function carries out common steps needed by create-volume for
+    # different mock-etcd configurations, docker configuration, create-volume
+    # requests and checking of responses for success/failure
+    def run_test(self, test_case):
+        self._test_case = test_case
+        # This is important to set as it is used by the base class to
+        # take decision which driver to instantiate
+        self._protocol = test_case.protocol
+        operation = 'volumedriver_create'
+        self._test_operation(operation)
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+    def override_configuration(self, config):
+        pass
+
+
+# class TestCreateVolumeDefault(CreateVolumeUnitTest, testtools.TestCase):
+class TestCreateVolumeDefault(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+
+# Provisioning = Full
+class TestCreateThickVolume(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {'provisioning': data.FULL}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+
+# Provisioning = Dedup
+class TestCreateDedupVolume(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {'provisioning': data.DEDUP}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+
+# FlashCache = True
+class TestCreateVolumeWithFlashCache(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+        # Return CPG dict without 'domain' member
+        # cpg = self.client.getCPG(cpg_name)
+
+
+# FlashCache = True
+class TestCreateVolumeFlashCacheAddToVVSFails(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
```

Comment:
Assert calls are missing

Reply:
Incomplete implementation of create volume unit tests. Will take care of this.